base/complex.jl

```diff
@@ -62,21 +62,29 @@ complex(z::Complex) = z
 flipsign(x::Complex, y::Real) = ifelse(signbit(y), -x, x)
 
 function show(io::IO, z::Complex)
-    r, i = reim(z)
+    if unitless(z) != z
```

Comment:
The use case here is to print `Complex{Unit}` differently? Why not just define a method `show{T<:Unit}(io::IO, z::Complex{T}`?